### PR TITLE
docs: add telemetry doc

### PR DESCRIPTION
**Pull request by paomian**

### CODE REVIEW

File: docs/user-guide/operations/configuration.md [hunk 1]
```diff
@@ -451,3 +451,25 @@ The environment variable also accepts list that are separated by a comma `,`, fo
 ```
 GREPTIMEDB_METASRV__META_CLIENT_OPTIONS__METASRV_ADDRS=127.0.0.1:3001,127.0.0.1:3002,127.0.0.1:3003
 ```
+
+### GreptimeDB Telemetry
+
+To make us better. GreptimeDB will collect some telemetry data, such as the version of GreptimeDB, the number of nodes, the OS of env, the arch of env, etc. We will not collect any user data, such as the database name, table name, query content, etc. It can be enabled or disabled by the compile feature flags `greptimedb-telemetry` in the build params.
```
Comment (xtang) [inline docs/user-guide/operations/configuration.md:457]:
```suggestion
To enhance our service, GreptimeDB collects certain telemetry data. This includes information like the GreptimeDB version, the number of nodes, the operating system used, the environment's architecture, and similar technical details. However, we respect your privacy and make sure not to collect any user-specific data, which entails database names, table names, query content, and the like.

This telemetry collection can easily be managed according to your preferences. You may choose to enable or disable it through the greptimedb-telemetry compile feature flags in the build parameters. Your experience and privacy are our top priority.
```

**Review by xtang — APPROVED** [review]

**Comment by xtang** [comment]
I have changed the first paragraph with the help of GPT, and the rest looks good to me.

**Comment by paomian** [comment]
> Can you give a shell example about how to enable or disable the `greptimedb-telemetry` flag? thx

Default flag is not include this feature. I can provide shell command which is enable `greptimedb-telemetry` flags.

```
cargo build --bin greptime --features 'greptimedb-telemetry'
```

The above command will enable our telemetry when compiling GreptimeDB.

**Comment by paomian** [comment]
I've changed some things in the section _How to disable telemetry?_ PTAL again.

**Comment by nicecui** [comment]
Waiting for the release of this feature...

**Comment by killme2008** [comment]
@nicecui Already merged, i think we can publish this doc.

**Review by nicecui — APPROVED** [review]

**Pull request merged**